op-node/rollup/types.go

```diff
@@ -254,6 +267,77 @@ func (cfg *Config) CheckL2GenesisBlockHash(ctx context.Context, client L2Client)
 	return nil
 }
 
+// dependencySetActivation stores sorted timestamp-dependencies pairs
+type dependencySetActivation struct {
+	timestamp uint64
+	chainIDs  []*big.Int
+}
+
+var (
+	// Store activations in sorted order
+	dependencySetActivations []dependencySetActivation
+	// Track latest activated dependency update position
+	latestActivationIndex int = -1
+)
+
+// IsDependencySetUpdate checks for any new chain dependencies that activate between
+// the previous and next block timestamps. Returns nil if no new dependencies activate,
+// or a slice of chain IDs for newly activated dependencies.
+func (cfg *Config) IsDependencySetUpdate(previousBlockTimestamp, nextBlockTimestamp uint64) []*big.Int {
+	if cfg.ClusterConfig == nil {
+		panic("missing cluster config")
+	}
```

Comment:
There are 2 approaches to writing tests for this code. We can either break out the implementation logic into smaller functions and test those functions individually or test `Config.IsDependencySetUpdate`. The tests should live in `types_test.go`

Reply:
should tests be included in this experimental branch?